site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md

```diff
@@ -0,0 +1,326 @@
+---
+title: "Automate checking ASVS controls using ZAP scripts"
+description: "Automate checking ASVS controls using ZAP scripts"
+summary: "Write scripts in ZAP which will check a target application's compliance against ASVS controls."
+type: post
+tags:
+- blog
+date: "2021-02-10"
+addSocialPreview: true
+authors:
+ - BlazingWind
+---
+
+Many security teams are required to provide security insights, and
+levels, of web applications they own. Security levels, or security
+posture, change dynamically based on the attack surface, known
+vulnerabilities, and numerous other factors. Due to this dynamic nature,
+assessing a security level is a challenging task, which needs to be
+performed on a regular basis and usually undertaken by comparing current
+implementation against controls defined in a standard, such as [OWASP
+Application Security Verification
+Standard](https://owasp.org/www-project-application-security-verification-standard/).
+
+The ASVS Standard spans 14 sections and over 260 controls, each assigned
+a level: level 1 that all applications should have, level 2 for
+applications processing sensitive data, and level 3 for critical
+applications. Level 1 controls are assessed to be fully
+penetration-testable from a black box perspective and this article
+focuses only on level 1 controls. Considering how broad ASVS is, it is
+difficult to ensure that every nook and cranny of an application is
+compliant with ASVS controls.
+
+And here raises a question: how to ensure more coverage of an
+application and make auditing applications with OWASP ASVS easier,
+faster, and more effective?
+
+# Enter OWASP ZAP and Its Scripting Engine
```

Comment:
These are better as ## so they appear in the toc

Reply:
Changed # to ## for all titles in the new commit